src/lib/analytics.js

```diff
@@ -172,6 +172,49 @@ function getNavigationType() {
   return '(not set)';
 }
 
+function logPrerenders() {
+  try {
+    /* The below complicated bit of code basically checks it's in this format:
+     *
+     * <script type="speculationrules">
+     *   {
+     *     "prerender": [
+     *       {
+     *         "source": "list",
+     *        "urls": ["url1", url2]
+     *       }
+     *     ],
+     *   }
+     * </script>
+     *
+     * And then extracts the unique URLs
+     */
+    const prerenderURLs = new Set(
+      [...document.querySelectorAll('script[type="speculationrules"]')]
+        .map((script) => JSON.parse(script.innerHTML))
+        .map((json) => json.prerender)
+        .filter((rules) => rules.map((rule) => rule.source === 'list'))
+        .map((rules) => rules.map((rule) => rule.urls))
+        .flat()
+        .flat(),
+    );
```

Comment:
Is there any better way of writing this?

Basically want to extract all URLs for multiple instances of this:

```html
      <script type="speculationrules">
        {
          "prerender": [
            {
              "source": "list",
             "urls": ["url1", "url2"]
            }
          ],
        }
      </script>
```

Reply:
I'd add a helper function that gets the full list of `prerender` rules. I tested this one out and it seems to work well:

```js
function getPrerenderRules() {
  return [...document.querySelectorAll('script[type=speculationrules]')]
    .map((s) => {
      try {
        return JSON.parse(s.textContent).prerender;
      } catch {
        // Ignore parse errors.
      }
    })
    .flat()
    .filter((rule) => rule && rule.source === 'list');
}
```

Then in your `logPrerenders()` function you could further flatted to get the list of URLs (note this uses `Set()` to remove duplicates):

```js
new Set(getPrerenderRules().map((r) => r.urls).flat())
```

Reply:
Great - thanks. Added.